Remotion/Validation/UnitTests/Validators/NotEmptyCollectionValidatorTest.cs

```diff
@@ -0,0 +1,218 @@
+    {
+      var propertyValidatorContext = CreatePropertyValidatorContext(new GenericCollection<string> { "someValue" });
+      var validator = new NotEmptyCollectionValidator(new InvariantValidationMessage("Fake Message"));
+
+      var validationFailures = validator.Validate(propertyValidatorContext).ToArray();
+
+      Assert.That(validationFailures, Is.Empty);
+    }
+
+    [Test]
+    public void Validate_WithPropertyValueIsNonEmptyReadOnlyCollection_ReturnsNoValidationFailures ()
+    {
+      var propertyValidatorContext = CreatePropertyValidatorContext(new ReadOnlyCollection<string> { "someValue" });
+      var validator = new NotEmptyCollectionValidator(new InvariantValidationMessage("Fake Message"));
+
+      var validationFailures = validator.Validate(propertyValidatorContext).ToArray();
+
+      Assert.That(validationFailures, Is.Empty);
+    }
+
+    [Test]
+    public void Validate_WithPropertyValueIsEmptyString_ReturnsNoValidationFailures ()
+    {
+      var propertyValidatorContext = CreatePropertyValidatorContext(string.Empty);
+      var validator = new NotEmptyCollectionValidator(new InvariantValidationMessage("Fake Message"));
+
+      var validationFailures = validator.Validate(propertyValidatorContext).ToArray();
+
+      Assert.That(validationFailures, Is.Empty);
+    }
+
+    [Test]
+    public void Validate_WithPropertyValueIsEmptyBinary_ReturnSingleValidationFailure ()
+    {
+      var propertyValidatorContext = CreatePropertyValidatorContext(Array.Empty<byte>());
+      var validator = new NotEmptyCollectionValidator(new InvariantValidationMessage("Fake Message"));
+
+      var validationFailures = validator.Validate(propertyValidatorContext).ToArray();
+
+      Assert.That(validationFailures.Length, Is.EqualTo(1));
+      Assert.That(validationFailures[0].ErrorMessage, Is.EqualTo("The value must not be empty."));
+    }
+
+    [Test]
+    public void Validate_WithPropertyValueIsObject_ReturnsSingleValidationFailure ()
+    {
+      var propertyValidatorContext = CreatePropertyValidatorContext(new object());
+      var validator = new NotEmptyCollectionValidator(new InvariantValidationMessage("Fake Message"));
+
+      var validationFailures = validator.Validate(propertyValidatorContext).ToArray();
+
+      Assert.That(validationFailures, Is.Empty);
+    }
+
+    private class Enumerable<T> : IEnumerable<T>
+    {
+      private readonly IEnumerable<T> _enumerable = Enumerable.Empty<T>();
+
+      public IEnumerator<T> GetEnumerator () => _enumerable.GetEnumerator();
+
+      IEnumerator IEnumerable.GetEnumerator () => GetEnumerator();
+    }
+
+    private class SimpleCollection : ICollection
+    {
+      private readonly ArrayList _list = new();
+
+      public IEnumerator GetEnumerator () => _list.GetEnumerator();
+
+      public void CopyTo (Array array, int index)
+      {
+        _list.CopyTo(array, index);
+      }
```

Comment:
that works. great. I'll approve the PR, though recommend you make this into an expression. will remained approved.